plugin/trino-fission-functions/src/main/java/io/trino/plugin/fission/FissionFunctionFunctions.java

```diff
@@ -40,21 +41,62 @@ private FissionFunctionFunctions()
 
     @ScalarFunction("fission_dnsdb")
     @Description("Converts the string to alternating case")
-    @SqlType(StandardTypes.VARCHAR)
-    public static Slice dnsdb(@SqlType(StandardTypes.VARCHAR) Slice slice)
+    @SqlType(StandardTypes.BIGINT)
+    public static long dnsdb(@SqlType(StandardTypes.VARCHAR) Slice slice) throws JSONException, IOException
     {
         CloseableHttpClient httpClient;
         HttpClientBuilder builder = HttpClientBuilder.create().setDefaultCookieStore(cookieStore);
         httpClient = builder.build();
         String result = "";
+        int count = 0;
         try {
             HttpGet getRequest = new HttpGet(String.format("%s/dnsdb?lookup=%s", FissionFunctionConfigProvider.getFissionFunctionBaseURL(), slice.toStringUtf8()));
             HttpResponse response = httpClient.execute(getRequest);
             result = EntityUtils.toString(response.getEntity());
+            JSONObject myObject = new JSONObject(result);
+
+            if (myObject.has("Message")) {
+                count = 0;
+            }
+            else if (myObject.has("total_count")) {
+                count = Integer.parseInt(myObject.getString("total_count"));
+            }
+        }
+        catch (IOException e) {
+            throw e;
+        }
+        catch (JSONException e) {
+            throw e;
+        }
+        return count;
+    }
+
+    @ScalarFunction("fission_despicablename")
+    @Description("Converts the string to alternating case")
+    @SqlType(StandardTypes.BIGINT)
+    public static long despicablename(@SqlType(StandardTypes.VARCHAR) Slice slice) throws JSONException, IOException
+    {
+        CloseableHttpClient httpClient;
```

Comment:
Would it make sense to refactor this a bit so that the logic to get "myObject" is all encapsulated in a method that can be reused for both these other methods?

Reply:
I would go one step further. `CloseableHttpClient` implements `AutoCloseable`, which means it should probably be managed using a try-with-resource. Also, these methods could easily be refactored to make use of a `executeFissionFunction` that would take the endpoint as a parameter and return the `jsonObject`.